darts/explainability/shap_explainer.py

```diff
@@ -0,0 +1,690 @@
+                    foreground_future_covariates,
+                ) = self.model.generate_predict_encodings(
+                    n=len(foreground_series) - self.model.min_train_series_length,
+                    series=foreground_series,
+                    past_covariates=foreground_past_covariates,
+                    future_covariates=foreground_future_covariates,
+                )
+
+        foreground_series = series2seq(foreground_series)
+        foreground_past_covariates = series2seq(foreground_past_covariates)
+        foreground_future_covariates = series2seq(foreground_future_covariates)
+
+        horizons, target_names = self._check_horizons_and_targets(
+            horizons, target_names
+        )
+
+        shap_values_list = []
+
+        for idx, foreground_ts in enumerate(foreground_series):
+
+            foreground_past_cov_ts = None
+            foreground_future_cov_ts = None
+
+            if foreground_past_covariates:
+                foreground_past_cov_ts = foreground_past_covariates[idx]
+
+            if foreground_future_covariates:
+                foreground_future_cov_ts = foreground_future_covariates[idx]
+
+            foreground_X = self.explainers._create_regression_model_shap_X(
+                foreground_ts,
+                foreground_past_cov_ts,
+                foreground_future_cov_ts,
+            )
+
+            shap_ = self.explainers.shap_explanations(
+                foreground_X, horizons, target_names
+            )
+
+            shap_values_dict = {}
+            for h in horizons:
+                tmp = {}
+                for t in target_names:
+                    tmp[t] = TimeSeries.from_times_and_values(
+                        shap_[h][t].time_index,
+                        shap_[h][t].values,
+                        columns=shap_[h][t].feature_names,
+                    )
+                shap_values_dict[h] = tmp
+
+            shap_values_list.append(shap_values_dict)
+
+        if len(shap_values_list) == 1:
+            shap_values_list = shap_values_list[0]
+
+        return ExplainabilityResult(shap_values_list)
+
+    def summary_plot(
+        self,
+        horizons: Optional[Sequence[int]] = None,
+        target_names: Optional[Sequence[str]] = None,
+        nb_samples: Optional[int] = None,
+        plot_type: Optional[str] = "dot",
+        **kwargs,
+    ):
+        """
+        Display a shap plot summary per (target, horizon)).
+        We here reuse the initial background data as foreground (potentially sampled) to give a general importance
+        plot for each feature.
+        If no target names and/or no horizons are provided, we plot all summary plots.
+
+        Parameters
+        ----------
+        horizons
+            Optionally, a list of integers representing which points/steps in the future we want to explain,
+            starting from the first prediction step at 0. Currently, only forecasting models are supported which
```

Comment:
Is this sentence necessary? As anyway the whole class is limited to RegressionModels.